lib/patch-dom.js

```diff
@@ -0,0 +1,73 @@
+import diff, {CREATE, UPDATE, MOVE, REMOVE} from 'dift';
+
+const isIframeEmbed = (elm) => elm && elm.tagName &&
+  elm.tagName.toLowerCase() === 'figure' && elm.querySelector('iframe');
+
+const updateIframeEmbed = ({oldFigure, newFigure}) => {
+  // We need to update the figure without touching the iframe
+  // to avoid reloading it for every update.
+  const oldCaption = oldFigure.querySelector('figcaption');
+  const newCaption = newFigure.querySelector('figcaption');
+
+  if (oldCaption && newCaption) {
+    oldFigure.replaceChild(newCaption, oldCaption);
+  } else if (oldCaption) {
+    oldFigure.removeChild(oldCaption);
+  } else if (newCaption) {
+    oldFigure.appendChild(newCaption);
+  }
+
+  const oldAttributes = Array.from(oldFigure.attributes);
+  const newAttributes = Array.from(newFigure.attributes);
+
+  oldAttributes.forEach(({name}) => {
+    oldFigure.removeAttribute(name);
+  });
+
+  newAttributes.forEach(({name, value}) => {
+    oldFigure.setAttribute(name, value);
+  });
+};
+
+const getKey = (elm) => {
+  if (isIframeEmbed(elm)) {
+    const iframe = elm.querySelector('iframe');
+    return iframe.src !== 'javascript:false' ? iframe.src : iframe.id;
+  }
+
+  return elm.outerHTML;
+};
+
+export default ({oldArticleElm, newArticleElm}) => {
+  const newList = Array.from(newArticleElm.childNodes);
+  const oldList = Array.from(oldArticleElm.childNodes);
+
+  const patch = (type, prev, next, pos) => {
+    switch (type) {
+      case CREATE:
+        if (next) {
+          oldArticleElm.insertBefore(next, oldArticleElm.childNodes[pos] || null);
+        }
+        break;
+      case UPDATE:
+        if (isIframeEmbed(prev)) {
+          updateIframeEmbed({
+            oldFigure: oldArticleElm.childNodes[pos],
+            newFigure: next
+          });
+        }
+        break;
+      case MOVE:
+        oldArticleElm.insertBefore(next, oldArticleElm.childNodes[pos]);
+        prev.remove();
```

Comment:
hm, not sure. might be another way of doing it but tests are failing without prev.remove() here.